### PR TITLE
Add Library templete

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**

### CODE REVIEW

File: docs/spec/sanitations.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+_Author_:  <!-- TODO: Add author name --> \
```
Comment (NipunaRanasinghe) [inline docs/spec/sanitations.md]: We can completely remove the `docs/` directory

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Lines starting with '#' are comments.
+# Each line is a file pattern followed by one or more owners.
+
+# See: https://help.github.com/articles/about-codeowners/
+
+# TODO: Add code owners
+# These owners will be the default owners for everything in the repo.
```
Comment (NipunaRanasinghe) [inline .github/CODEOWNERS]: Missing newline at the end of file. Please update all the similar files

File: docs/spec/sanitations.md (hunk 1 above)
Comment (DharshiBalasubramaniyam) [reply docs/spec/sanitations.md]: Done!

File: .github/CODEOWNERS (hunk 2 above)
Comment (DharshiBalasubramaniyam) [reply .github/CODEOWNERS]: Done!

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

File: settings.gradle [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*
```
Comment (ThisaruGuruge) [inline settings.gradle:1]: Let's replace this license header with the WSO2 license header.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (ThisaruGuruge) [inline gradle/wrapper/gradle-wrapper.properties]: Shall we use Gradle 8.11.1 here? We _might_ have to replace the wrapper JAR as well.

File: settings.gradle (hunk 3 above)
Comment (DharshiBalasubramaniyam) [reply settings.gradle:1]: done!

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (DharshiBalasubramaniyam) [reply gradle/wrapper/gradle-wrapper.properties]: done!

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**